File: AEDS/C#/Agenda em CSharp/PessoaJuridica.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.RegularExpressions;

namespace Agenda_em_CSharp
{
    class PessoaJuridica : Pessoa
    {

        private static readonly string validador = "^[0-9]{2}\\.?[0-9]{3}\\.?[0-9]{3}\\/?[0-9]{4}\\-?[0-9]{2}$";

        public PessoaJuridica(string numero){

            if (ValidaCNPJ(numero)) this.numero = numero;

            else throw new ArgumentException("Invalid Format");
        }

        private bool ValidaCNPJ(string numero){

            if (Regex.IsMatch(numero, validador)) return true;

            else return false;
        }
        public override void SetNumero(string numero){

            if (ValidaCNPJ(numero)) this.numero = numero;

            else throw new ArgumentException("Invalid CNPJ");
        }
    }
}

File: AEDS/Java/Trabalho Agenda/src/Agenda/PessoaJuridica.java
package Agenda;

public class PessoaJuridica extends Pessoa {

    private static final String validacao = "^[0-9]{3}\\.?[0-9]{3}\\.?[0-9]{3}\\-?[0-9]{2}|[0-9]{2}\\.?[0-9]{3}\\.?[0-9]{3}\\/?[0-9]{4}\\-?[0-9]{2}$";

    public PessoaJuridica(String numero){

        if (ValidaCNPJ(numero) == true) this.CadastroPessoa = numero;

    }

    private boolean ValidaCNPJ(String numero) throws IllegalArgumentException{

        if (!numero.matches(validacao)){

            throw new IllegalArgumentException("Not a valid CNPJ format");
        }

        return true;
    }

    @Override
    public void setNumero(String CadastroPessoa) { this.CadastroPessoa = CadastroPessoa; }
}
